src/pagerduty/pagerduty_client.bal

```diff
@@ -0,0 +1,691 @@
+    # ```
+    #
+    # + escalationPolicy - The escalation policy to be created.
+    # + return - A `pagerduty:EscalationPolicy` or else a `pagerduty:Error` if any error  occurred
+    public remote function create(EscalationPolicy escalationPolicy) returns EscalationPolicy|Error {
+        if (self.emailId == "") {
+            return Error(message = "Can't access the `createEscalationPolicy` operation without logged-in user" +
+                         " email id", cause = getError());
+        }
+        return <@untainted> createEscalationPolicy(self.escalationPolicyClient, escalationPolicy, self.emailId);
+    }
+
+    # Gets informations about a given existing escalation policy and its rules.
+    # ```ballerina
+    # pagerduty:EscalationPolicy|pagerduty:Error response = escalationPolicyClient->getById(<POLICY_ID>);
+    # ```
+    #
+    # + escalationPolicyId - The escalation policy id, which to be update
+    # + return - A `pagerduty:EscalationPolicy` or else a `pagerduty:Error` if any error  occurred
+    public remote function getById(string escalationPolicyId) returns EscalationPolicy|Error {
+        return <@untainted> getEscalationPolicyById(self.escalationPolicyClient, escalationPolicyId);
+    }
+
+    # Updates an existing escalation policy and rules.
+    # ```ballerina
+    # pagerduty:EscalationPolicy|pagerduty:Error response = escalationPolicyClient->update(<POLICY_ID>,
+    #                                                                                       <ESCALATION_POLICY>);
+    # ```
+    #
+    # + escalationPolicyId - The escalation policy id, which to be update
+    # + escalationPolicy - The escalation policy details which to be update
+    # + return - A `pagerduty:EscalationPolicy` or else a `pagerduty:Error` if any error  occurred
+    public remote function update(string escalationPolicyId, EscalationPolicy escalationPolicy)
+                                 returns EscalationPolicy|Error {
+        return <@untainted> updateEscalationPolicy(self.escalationPolicyClient, escalationPolicyId, escalationPolicy);
+    }
+
+    # Removes an existing `EscalationPolicy`
+    # ```ballerina
+    # pagerduty:Error? response = escalationPolicyClient->delete(<POLICY_ID>);
+    # ```
+    #
+    # + escalationPolicyId - The escalation policy id, which to be update
+    # + return - A `pagerduty:Error` if any error occurred or `nil`
+    public remote function delete(string escalationPolicyId) returns Error? {
+        string path = ESCALATION_POLICES_PATH + BACK_SLASH + escalationPolicyId;
+        return <@untainted> delete(self.escalationPolicyClient, path);
+    }
+
+};
+
+# The `pagerduty:ScheduleClient` used to create/get/delete the schedule.
+#
+# + scheduleClient - The `pagerduty:ScheduleClient`
+# + emailId - The email id for the logged-in user
+public type ScheduleClient client object {
+
+    private http:Client scheduleClient;
+    public string emailId = "";
```

Comment:
Is it possible to make this `emailId` field private and assign it through the constructor?
This `emailId` field should not allow the user to edit after creation. 
Check other clients as well